flink-core/src/test/java/org/apache/flink/api/common/eventtime/AscendingTimestampsWatermarksTest.java

```diff
@@ -30,7 +30,10 @@
 	@Test
 	public void testWatermarkBeforeRecords() {
 		final TestingWatermarkOutput output = new TestingWatermarkOutput();
-		final AscendingTimestampsWatermarks<Object> watermarks = new AscendingTimestampsWatermarks<>();
```

Comment:
Do we need to change these?
The `AscendingTimestampsWatermarksTest` should test the `AscendingTimestampsWatermarks` class, and not be coupled to the `WatermarkStrategy` class.

An indicator that the scope of testing is off here is the `() -> null` lambda needed to instantiate the class to be tested, even though it has nothing to do with the tests.

Reply:
You're right, I did change that to see if the builders work because there were no tests for that before, I did add tests in `WatermarkStrategyTest` already so I will undo these changes.